### ISSUE TITLE
AC Migration allows duplicate column names when using t.belongs_to

### ISSUE THREAD

**Issue by neidiom** [rails/rails#33024]
I just had a migration similar to the one below, resource_id will be created only once, I think this should generate a warning at least.

````
class CreateAlerts < ActiveRecord::Migration[5.2]
  def change
    create_table :alerts do |t|
      t.belongs_to :resource, index: true
      t.string :resource_id
      t.string :resource_group_name
      t.timestamps
    end
  end
end
````

**Comment by fedxgibson** [comment]
What if we abort the migration? 

One may miss the warning, and start working on features, developing code, designing, and so on assuming the two columns exists (It should break in the tests).

Would it be better to save developer's time and let him know it didn't work from the beginning?

**Comment by rafaelfranca** [comment]
Yeah. We should just raise if we try to add the same column on the same migrations more than once.

**Comment by fedxgibson** [comment]
I created a pull request for this:

https://github.com/rails/rails/pull/33029

I'm new in rails contributions so any feedback would be appreciated.

**Comment by fedxgibson** [comment]
@neidiom can you check this PR?

**Comment by kamipo** [comment]
Fixed by #33029.
